drivers/rf-transceiver/hmc6300/hmc6300.c

```diff
@@ -0,0 +1,439 @@
+	}
+	no_os_gpio_set_value(dev->en, NO_OS_GPIO_HIGH);
+
+	return 0;
+}
+
+/* Write the write-accessible registers of the register map. */
+int hmc6300_write_regmap(struct hmc6300_dev *dev, const uint8_t *regmap)
+{
+	int ret, r;
+	uint8_t reg;
+
+	for (r = 1; r < 24; r++) {
+		if (r >= 13 && r <= 15)
+			continue;
+		reg = regmap[r];
+		ret = hmc6300_write_row(dev, r, reg);
+		if (ret < 0)
+			return ret;
+		ret = hmc6300_read_row(dev, r, &reg);
+		if (ret < 0)
+			return ret;
+		if (reg != regmap[r])
+			return -EFAULT;
+	}
+
+	return 0;
+}
+
+/* Write a device parameter (bit or bit-field of a certain row). */
+int hmc6300_write(struct hmc6300_dev *dev, uint16_t param, uint8_t value)
+{
+	int ret;
+	uint8_t reg;
+	uint8_t row = HMC6300_ROW(param);
+	uint8_t mask = HMC6300_MASK(param);
+
+	ret = hmc6300_read_row(dev, row, &reg);
+	if (ret)
+		return ret;
+
+	reg &= ~mask;
+	reg |= no_os_field_prep(mask, value);
+	return hmc6300_write_row(dev, row, reg);
+}
+
+/* Read a device parameter (bit or bit-field of a certain row). */
+int hmc6300_read(struct hmc6300_dev *dev, uint16_t param, uint8_t *value)
+{
+	int ret;
+	uint8_t reg;
+
+	if (!value)
+		return -EINVAL;
+
+	ret = hmc6300_read_row(dev, HMC6300_ROW(param), &reg);
+	if (ret)
+		return ret;
+
+	*value = no_os_field_get(HMC6300_MASK(param), reg);
+	return 0;
+}
+
+/* Read the whole register map and store it into a 32-byte output buffer. */
+int hmc6300_read_regmap(struct hmc6300_dev *dev, uint8_t *regmap)
+{
+	int r, ret;
+
+	for (r = 0; r < 32; r++) {
+		ret = hmc6300_read_row(dev, r, &regmap[r]);
+		if (ret)
+			break;
+	}
+
+	return ret;
+}
+
+/* Enable temperature sensor. */
+int hmc6300_enable_temp(struct hmc6300_dev *dev, bool enable)
+{
+	int ret = hmc6300_write(dev, HMC6300_EN_TEMPFLASH, enable);
+	if (ret)
+		return ret;
+
+	return hmc6300_write(dev, HMC6300_TEMPSENSOR_PWRDN, !enable);
+}
+
+/* Get a temperature reading. */
+int hmc6300_get_temp(struct hmc6300_dev *dev, uint8_t *temp)
+{
+	return hmc6300_read(dev, HMC6300_TEMPS, temp);
+}
+
+/* Enable FSK/MSK modulation inputs. */
+int hmc6300_enable_fm(struct hmc6300_dev *dev, bool enable)
+{
+	return hmc6300_write(dev, HMC6300_ENABLE_FM, (uint8_t)enable);
+}
+
+/* Power On/Off the chip. */
+int hmc6300_enable(struct hmc6300_dev *dev, bool enable)
+{
+	uint8_t row4 = enable ? 0 : 0xff;
+	/*
+	This collectively turns on/off the following:
+		HMC6300_DRIVER_PWRDN
+		HMC6300_UPMIXER_PWRDN
+		HMC6300_IFVGA_PWRDN
+		HMC6300_DIVIDER_PWRDN
+		HMC6300_PA_PWRDN
+		HMC6300_RFVGA_PWRDN
+		HMC6300_TRIPLER_PWRDN
+		HMC6300_IF_UPMIXER_PWRDN
+	*/
+	int ret = hmc6300_write_row(dev, 4, row4);
+	if (ret)
+		return ret;
+
+	return hmc6300_write(dev, HMC6300_EN_SYNTH_LDO, enable);
+}
+
+/* Set the IF attenuation in steps from 0 (highest gain) to 15.
+ * For digital setting to dB correlation, see Figure 7 of datasheet.
+ */
+int hmc6300_set_if_attn(struct hmc6300_dev *dev, uint8_t attn)
+{
+	return hmc6300_write(dev, HMC6300_IFVGA_VGA_ADJ, attn);
```

Comment:
same here: we can use the write function inline instead for adding extra wrapper.